src/execCmd.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { join } from 'path';
+import { inspect } from 'util';
+import { fs as fsCore } from '@salesforce/core';
+import { Duration, env, parseJson } from '@salesforce/kit';
+import { AnyJson, isNumber } from '@salesforce/ts-types';
+import Debug from 'debug';
+import * as shelljs from 'shelljs';
+import { ExecCallback, ExecOptions, ShellString } from 'shelljs';
+
+export interface ExecCmdOptions extends ExecOptions {
+  /**
+   * Throws if this exit code is not returned by the child process.
+   */
+  ensureExitCode?: number;
+}
+
+export interface ExecCmdResult {
+  /**
+   * Command output from the shell.
+   *
+   * @see https://www.npmjs.com/package/shelljs#execcommand--options--callback
+   */
+  shellOutput: ShellString;
+
+  /**
+   * Command output parsed as JSON, if `--json` param present.
+   */
+  jsonOutput?: AnyJson;
+
+  /**
+   * The JsonParseError if parsing failed.
+   */
+  jsonError?: Error;
+
+  /**
+   * Command execution duration.
+   */
+  execCmdDuration: Duration;
+}
+
+const DEFAULT_SHELL_OPTIONS = {
+  timeout: 300000, // 5 minutes
+  cwd: process.cwd(),
+  env: Object.assign({}, process.env),
+  silent: true,
+};
+
+// Create a Duration instance from process.hrtime
+const hrtimeToMillisDuration = (hrTime: [number, number]) =>
+  Duration.milliseconds(hrTime[0] * Duration.MILLIS_IN_SECONDS + hrTime[1] / 1e6);
+
+// Add JSON output if json flag is set
+const addJsonOutput = (cmd: string, result: ExecCmdResult): ExecCmdResult => {
+  if (cmd.includes('--json')) {
+    try {
+      result.jsonOutput = parseJson(result.shellOutput.stdout);
```

Comment:
this is typically where I've had color problems.  
https://github.com/mshanemc/plugin-helpers/blob/ea8d6bdc13833f29bb761b95ed7de64ba0dc96a0/src/execProm.ts#L11

Reply:
What reproduces the problem?  I'm happy to change it if needed.

Reply:
This can be caused by colors being injected directly into the result instead of delegating color application to ux functions. Question is, should these be stripped w/o considering intent.

Reply:
That seems like a bad practice to me.  Necessarily injecting colors into json output should blow up a test imo.  Colors are for human output, not json.

Reply:
I agree @shetzel, not a good practice. I am ok with not stripping terminal characters from JSON and letting it fail on parse.